### ISSUE TITLE
[feature request] track git branch

### ISSUE THREAD

**Issue by Sibz** [Jannchie/codetime-web-v3#8]
Thank you for this great add on!

I would like to see time spent on branches within a repository/project - I typically do one branch per feature/fix, so this would enable me to account for time at the task level which would be enormously beneficial.

**Comment by Jannchie** [comment]
That sounds like a great idea, I'll give it a try and add this feature.

**Comment by JackBlair87** [comment]
I agree this would be an incredible addition! Codetime is absolutely goated.

**Comment by Jannchie** [comment]
We have successfully implemented a basic feature to track git branches. 

https://codetime.dev/en/dashboard/workspace

![image](https://github.com/Jannchie/codetime-web-v3/assets/29743310/8d3c5cd5-f07f-4e95-8a37-c80a0e7c4bba)

However, there are still many tasks to be completed. I have come to the realization that distinguishing projects based on workspace (folder name) is not ideal. A better approach would be to identify git origin URLs to merge code time directed to the same repository. 

Furthermore, the records before the launch of this feature cannot be updated and will currently show as “Unknown.” But I think it is possible to calculate previous code time by reading git logs.

We will continue to make updates to the relevant features.

**Comment by Sibz** [comment]
Fantastic!

I look forward the project by git origin feature to be introduced.
